verde/coordinates.py

```diff
@@ -770,15 +771,186 @@ def block_split(coordinates, spacing=None, adjust="spacing", region=None, shape=
+    [0 1 2 0 1 2 0 1 2]
+    >>> for dimension in indices[3]:
+    ...     print(dimension)
+    [2 2 2 3 3 3 4 4 4]
+    [2 3 4 2 3 4 2 3 4]
+    >>> # To get the coordinates for each window, use indexing
+    >>> print(coords[0][indices[0]])
+    [-5. -4. -3. -5. -4. -3. -5. -4. -3.]
+    >>> print(coords[1][indices[0]])
+    [6. 6. 6. 7. 7. 7. 8. 8. 8.]
+
+    If the coordinates are 1D, the indices will also be 1D:
+
+    >>> coords1d = [coord.ravel() for coord in coords]
+    >>> window_coords, indices = rolling_split(coords1d, size=2, spacing=2)
+    >>> print(len(indices[0]))
+    1
+    >>> print(indices[0][0])
+    [ 0  1  2  5  6  7 10 11 12]
+    >>> print(indices[1][0])
+    [ 2  3  4  7  8  9 12 13 14]
+    >>> print(indices[2][0])
+    [10 11 12 15 16 17 20 21 22]
+    >>> print(indices[3][0])
+    [12 13 14 17 18 19 22 23 24]
+    >>> # The returned indices can be used in the same way as before
+    >>> print(coords1d[0][indices[0]])
+    [-5. -4. -3. -5. -4. -3. -5. -4. -3.]
+    >>> print(coords1d[1][indices[0]])
+    [6. 6. 6. 7. 7. 7. 8. 8. 8.]
+
+    By default, the windows will span the entire data region. You can also
+    control the specific region you'd like the windows to cover:
+
+    >>> # Coordinates on a larger region but with the same spacing as before
+    >>> coords = grid_coordinates((-10, 5, 0, 20), spacing=1)
+    >>> # Get the rolling window indices but limited to the region from before
+    >>> window_coords, indices = rolling_split(
+    ...     coords, size=2, spacing=2, region=(-5, -1, 6, 10),
+    ... )
+    >>> # The windows should still be in the same place as before
+    >>> for coord in window_coords:
+    ...     print(', '.join(['{:.1f}'.format(i) for i in coord]))
+    -4.0, -2.0, -4.0, -2.0
+    7.0, 7.0, 9.0, 9.0
+    >>> # And indexing the coordinates should also provide the same result
+    >>> print(coords[0][indices[0]])
+    [-5. -4. -3. -5. -4. -3. -5. -4. -3.]
+    >>> print(coords[1][indices[0]])
+    [6. 6. 6. 7. 7. 7. 8. 8. 8.]
+
+    """
+    shapes = [coord.shape for coord in coordinates]
+    if not all(shape == shapes[0] for shape in shapes):
+        raise ValueError(
+            "Coordinate arrays must have the same shape. Given shapes: {}".format(
+                shapes
+            )
+        )
+    if region is None:
+        region = get_region(coordinates)
+    # Calculate the region spanning the centers of the rolling windows
+    window_region = [
+        dimension + (-1) ** (i % 2) * size / 2 for i, dimension in enumerate(region)
+    ]
+    centers = grid_coordinates(
+        window_region, spacing=spacing, shape=shape, adjust=adjust
+    )
+    # pykdtree doesn't support query_ball_point yet and we need that
+    tree = kdtree(coordinates, use_pykdtree=False)
+    indices1d = tree.query_ball_point(
+        np.transpose(n_1d_arrays(centers, 2)), r=size / 2, p=np.inf
+    )
+    indices = [np.unravel_index(i, shape=shapes[0]) for i in indices1d]
```

Comment:
This line is throwing an error if one of the rolling windows is unpopulated.
For example, if I run the following code:
```python
import verde as vd

coordinates = vd.scatter_points(region=(0, 100, 0, 100), size=30, random_state=1234)
vd.rolling_split(coordinates, size=20, spacing=10)
```

I get the following:
```traceback
Traceback (most recent call last):
  File "rolling.py", line 13, in <module>
    vd.rolling_split(coordinates, size=size, spacing=10)
  File "/home/santi/git/verde/verde/coordinates.py", line 953, in rolling_split
    indices = [np.unravel_index(i, shape=shapes[0]) for i in indices1d]
  File "/home/santi/git/verde/verde/coordinates.py", line 953, in <listcomp>
    indices = [np.unravel_index(i, shape=shapes[0]) for i in indices1d]
  File "<__array_function__ internals>", line 6, in unravel_index
TypeError: indices must be integral: the provided empty sequence was inferred as float. Wrap it with 'np.array(indices, dtype=np.intp)'
```

I think converting `i` to a `np.array` with `dtype=int` should solve it:
```suggestion
    indices = [
        np.unravel_index(np.array(i, dtype=int), shape=shapes[0])
        for i in indices1d
    ]
```

Reply:
Right, that seems reasonable. Thanks!